cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerHttpHandlerInternal.java

```diff
@@ -87,6 +95,29 @@ public void run(FullHttpRequest request, HttpResponder responder) {
     }
   }
 
+  @GET
+  @Path("/token")
+  public void token(FullHttpRequest request, HttpResponder responder) {
+    LOG.error("RECEIVED toke request...");
+    HttpURLConnection connection = null;
+    try {
+      URL url = new URL(METADATA_SERVER_GET_TOKEN_URL);
```

Comment:
You can use the `io.cdap.common.http.HttpRequests` class for simplification and better error handling

Reply:
Done!